src/meteors/lime.py

```diff
@@ -241,156 +252,453 @@ def to(self, device: torch.device) -> Self:
+        if isinstance(segment_name, str):
+            return segment_name
+        if isinstance(segment_name, list):
+            return tuple(segment_name)
+        raise ValueError(f"Incorrect segment {segment_name} type. Should be either a list or string")
 
     @staticmethod
-    def __get_band_mask_from_names_dict(image: Image, band_names: dict[tuple[str, ...] | str, int]) -> torch.Tensor:
-        grouped_band_names = Lime.__get_grouped_band_names(band_names)
+    def _extract_bands_from_spyndex(
+        segment_name: Iterable[str] | str,
+    ) -> tuple[str, ...]:
+        """Users may pass either band names or indices names, as in spyndex library. Index usually consists of multiple segments and needs to be expanded into a list of bands which can be then converted to waveband range."""
+        band_names_segment: list[str] = []
+
+        if isinstance(segment_name, str):
+            segment_name = tuple([segment_name])
+        for band_name in segment_name:
+            if band_name in spyndex.indices:
+                band_names_segment = band_names_segment + (spyndex.indices[band_name].bands)
+            elif band_name in spyndex.bands:
+                band_names_segment.append(band_name)
+            else:
+                raise ValueError(
+                    f"Invalid band name {band_name}, band name must be either in `spyndex.indices` or `spyndex.bands`"
+                )
+
+        return tuple(band_names_segment)
 
-        device = image.image.device
-        resolution_segments = Lime.__get_resolution_segments(image.wavelengths, grouped_band_names, device=device)
+    @staticmethod
+    def _get_band_ranges_wavelengths_from_band_names(
+        image: Image,
+        band_names: list[str | list[str]] | dict[tuple[str, ...] | str, int],
+    ) -> tuple[
+        dict[str | tuple[str, ...], list[tuple[float, float]]],
+        dict[str | tuple[str, ...], int],
+    ]:
+        """function extracts ranges of wavelengths from the band names
 
-        axis = [0, 1, 2]
-        axis.remove(image.band_axis)
+        Args:
+            image (Image): an image for which the band mask is created. The only important element of image are the wavelengths
+            band_names (list[str  |  list[str]] | dict[tuple[str, ...]  |  str, int]): list or dictionary with band names or segments
 
-        band_mask = resolution_segments.unsqueeze(axis[0]).unsqueeze(axis[1])
-        size_image = image.image.size()
-        size_mask = band_mask.size()
+        Returns:
+            dict[str | tuple[str, ...], list[tuple[float, float]]], dict[str | tuple[str, ...], int]: tuple containing the dictionary with mapping segment labels into wavelength ranges and mapping from segment labels into segment ids
+        """
 
-        repeat_dims = [s2 // s1 for s1, s2 in zip(size_mask, size_image)]
-        band_mask = band_mask.repeat(repeat_dims)
+        segments_list = []
```

Comment:
@Fersoil is there a need for empty list initialization if it is always assigned?